src/main/java/nl/tudelft/contextproject/gui/CreateScriptController.java

```diff
@@ -728,6 +467,268 @@ private void showErrorDialog(Exception e, File file) {
+                editCamera.setStyle(REDBORDER);
+                isValid = false;
+            }
+
+            if (editPreset.getSelectionModel().isEmpty()) {
+                editPreset.setStyle(REDBORDER);
+                isValid = false;
+            }
+
+            if (editDescription.getText().isEmpty()) {
+                editDescription.setStyle(REDBORDER);
+                isValid = false;
+            }
+
+            if (isValid) {
+                editCamera.setStyle("");
+                editPreset.setStyle("");
+                editDescription.setStyle("");
+
+                editConfirmAction(lastSelectedRow.get().getItem());
+            }
+        });
+
+        btnEditRemove.setOnAction(event -> {
+            Shot shot = lastSelectedRow.get().getItem();
+            tableEvents.getItems().remove(shot);
+
+            editDoneAction();
+        });
+    }
+
+    /**
+     * Sets listeners and actions on the table.
+     */
+    private void initTable() {
+        tableEvents.getSelectionModel().selectedItemProperty().addListener((obs, ov, nv) -> {
+            if (nv != null) {                
+                editShot.setText(nv.getShotId());
+                editCamera.getSelectionModel().select(nv.getCamera().getNumber());
+                if (nv.getPreset() != null) {
+                    editPreset.getSelectionModel().select(nv.getPreset().getId()); //+1
+                } else {
+                    editPreset.getSelectionModel().select(0);
+                }
+                editDescription.setText(nv.getDescription());
+
+                if (ov != nv && ov != null) {
+                    editDoneAction();
+                }
+            }
+        });
+
+        tableEvents.setOnMouseClicked(event -> {
+            if (event.getClickCount() == 2 && lastSelectedRow.get() != null) {
+                // Make sure the HBox is drawn over the selected row.
+                editBox.setLayoutY(lastSelectedRow.get().getLayoutY()
+                        + editBox.getTranslateY()
+                        + 40);
+
+                editBlockScroll.setVisible(true);
+                editBox.toFront();
+                editBox.setVisible(true);
+            }
+        });
+    }
+
+    /**
+     * Sets the action to be taken when an edit is complete.
+     * @param shot The shot that is edited.
+     */
+    private void editConfirmAction(Shot shot) {
+        Shot backup = new Shot(shot.getNumber(), shot.getShotId(), 
+                shot.getCamera(), shot.getPreset(), shot.getDescription());
+
+        shot.setShotId(editShot.getText());
+        shot.setCamera(Camera.getCamera(editCamera.getSelectionModel().getSelectedIndex()));
+        if (!editPreset.getSelectionModel().getSelectedItem().equals("None")) {
+            shot.setPreset(Camera.getCamera(editCamera.getSelectionModel().getSelectedIndex())
+                    .getPreset(new Integer(editPreset.getSelectionModel().getSelectedIndex()))); //-1
```

Comment:
The savefileFix branch fixes this, will be fixed when merge requests are dealt with